### PR TITLE
Created page which contains tracks of changes in database 

### LINKED ISSUE #389: As an admin, I want to view information about changed fields in database

### PR DESCRIPTION
**Pull request by VOLIMAX**
dev
## GitHub Project

* [Main GitHub Project ticket](https://github.com/EventsExpress/EventsExpress/projects/2)

## Code reviewers

- [ ] @iuriikhrystiuk 

### Second Level Review

- [ ] @sand0 

## Summary of issue

Needed to create a page, where you can track changes in database.

## Summary of change

Created a page, on which admin can track changes in database. Besides added pagination and a possibility to filter tracks in different ways. 

## Testing approach

Starting to create unit tests

## CHECK LIST
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: EventsExpress/ClientApp/.eslintcache [hunk 1]
```diff
@@ -0,0 +1 @@
...src\\actions\\tracks\\track-list-action.js",[],"C:\\Users\\User\\source\\repos\\EventsExpress\\EventsExpress\\ClientApp\\src\\services\\TrackService.js",[],"C:\\Users\\User\\source\\repos\\EventsExpress\\EventsExpress\\ClientApp\\src\\components\\tracks\\track.js",[],"C:\\Users\\User\\source\\repos\\EventsExpress\\EventsExpress\\ClientApp\\src\\components\\tracks\\track-list.js",[],"C:\\Users\\User\\source\\repos\\EventsExpress\\EventsExpress\\ClientApp\\src\\components\\tracks\\tracks-filter.js",[],"C:\\Users\\User\\source\\repos\\EventsExpress\\EventsExpress\\ClientApp\\src\\components\\tracks\\track-item.js",[],"C:\\Users\\User\\source\\repos\\EventsExpress\\EventsExpress\\ClientApp\\src\\constants\\changesTypeEnum.js",[],{"ruleId":null,"fatal":true,"severity":2,"message":"1351","line":25,"column":8},"Parsing error: Identifier 'AuthComponent' has already been declared\n\n  23 | import AuthComponent from \"../../security/authComponent\";\n  24 | import {Roles} from '../../constants/userRoles';\n> 25 | import AuthComponent from \"../../security/authComponent\";\n     |        ^\n  26 | import { Roles } from '../../constants/userRoles';\n  27 |\n  28 | const useStyles = useStyle;"]
```
Comment (iuriikhrystiuk) [inline EventsExpress/ClientApp/.eslintcache]: Please undo this file.

File: EventsExpress/ClientApp/src/components/event/event-cancel-modal.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
+﻿import React, { Component } from "react";
+import { connect } from 'react-redux';
+import DialogActions from "@material-ui/core/DialogActions";
+import Button from "@material-ui/core/Button";
+import Dialog from "@material-ui/core/Dialog";
+import { DialogContent } from '@material-ui/core';
+import { renderErrorMessage } from '../helpers/helpers';
+import { setEventCanelationModalStatus } from '../../actions/event-item-view';
+
+class EventCancelModal extends Component {
```
Comment (iuriikhrystiuk) [inline EventsExpress/ClientApp/src/components/event/event-cancel-modal.js]: I believe this file was removed? Please consider merging properly.

File: EventsExpress/ClientApp/src/components/event/event-filter-status.js [hunk 3]
```diff
@@ -36,7 +51,10 @@ class EventFilterStatus extends Component {
                                 }
                                 return input.onChange(newValue);
                             }} />
-                        {eventStatusHistoryReadingString(option)}
+                        {options.length >= 4
```
Comment (iuriikhrystiuk) [inline EventsExpress/ClientApp/src/components/event/event-filter-status.js]: Please, pass options from outside of this component, rather than doing unobvious logic inside.

File: EventsExpress/ClientApp/src/components/helpers/helpers.js [hunk 4]
```diff
@@ -218,6 +218,16 @@ export const renderDatePicker = ({ input: { onChange, value }, minValue, label }
     />
 }
 
+export const renderTracksDatePicker = ({ input: { onChange, value }, label }) => {
```
Comment (iuriikhrystiuk) [inline EventsExpress/ClientApp/src/components/helpers/helpers.js]: Why did you need a special datepicker? Don't we already have datepickers?

File: EventsExpress/ClientApp/src/components/event/simple-modal.js [hunk 5]
```diff
@@ -50,15 +50,17 @@ export default class SimpleModal extends Component {
...
-                            <Button
-                                fullWidth={true}
-                                type="button"
-                                value="Login"
-                                color="primary"
-                                onClick={this.onConfirm}
-                            >
-                                confirm
-                            </Button>
+                            {this.props.action &&
```
Comment (iuriikhrystiuk) [inline EventsExpress/ClientApp/src/components/event/simple-modal.js]: Why is this change required?

File: EventsExpress/ClientApp/src/components/tracks/track-item.js [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+                    </tr>
+                ))
+                return {
+                    changesTypeText,
+                    data
+                };
+        }
+    }
+
+    filter = values => {
```
Comment (iuriikhrystiuk) [inline EventsExpress/ClientApp/src/components/tracks/track-item.js]: I don't see this method used anywhere. Please remove it if so.

File: EventsExpress/ClientApp/src/components/tracks/track.js [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+import {connect} from 'react-redux';
+
+class Tracks extends Component {
+
+    componentDidMount = () => {
+        this.props.getAllTracks(this.props.tracks.filter);
+        this.props.getEntityNames();
+    }
+
+    handleSubmit = async (filters) => {
```
Comment (iuriikhrystiuk) [inline EventsExpress/ClientApp/src/components/tracks/track.js:23]: Please merge handle submit and handle page change into single method, since logic is duplicated.

File: EventsExpress/ClientApp/src/containers/add-comment.js [hunk 8]
```diff
@@ -7,6 +7,10 @@ import { reset } from 'redux-form';
 
 class CommentWrapper extends React.Component {
 
+    submit = values => {
```
Comment (iuriikhrystiuk) [inline EventsExpress/ClientApp/src/containers/add-comment.js:10]: Why was this added?

File: EventsExpress/ClientApp/src/containers/tracks/tracks.js [hunk 9]
```diff
@@ -0,0 +1,11 @@
+import Tracks  from '../../components/tracks/track';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+
+const mapStateToProps = (state) => ({
+    tracks: state.tracks
+});
+
+const mapDispatchToProps = (dispatch) => { return bindActionCreators(() => { }, dispatch); };
```
Comment (iuriikhrystiuk) [inline EventsExpress/ClientApp/src/containers/tracks/tracks.js]: If there are no dispatchers, just pass null.

File: EventsExpress/ClientApp/src/store/initialState.js [hunk 10]
```diff
@@ -135,6 +135,22 @@ const initialState = {
...
         filter: eventHelper.getDefaultEventFilter(),
     },
+    tracks: {
+        isPending: false,
+        isError: false,
+        data: {
+            items:{},
+            pageViewModel: {},
+        },
+        filter: {
```
Comment (iuriikhrystiuk) [inline EventsExpress/ClientApp/src/store/initialState.js]: You don't need filter object anymore.

File: EventsExpress/Controllers/TracksController.cs [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+using EventsExpress.Core.DTOs;
+using EventsExpress.Core.IServices;
+using EventsExpress.ViewModels;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+
+namespace EventsExpress.Controllers
+{
+    [Route("api/[controller]/[action]")]
+    [Authorize(Roles = "Admin")]
```
Comment (iuriikhrystiuk) [inline EventsExpress/Controllers/TracksController.cs]: Please use policies here.

File: EventsExpress/ClientApp/package.json [hunk 12]
```diff
@@ -9,6 +9,7 @@
         "@material-ui/lab": "4.0.0-alpha.24",
         "@material-ui/pickers": "3.2.0",
         "@material-ui/styles": "4.2.0",
+        "@reduxjs/toolkit": "^1.5.0",
```
Comment (iuriikhrystiuk) [inline EventsExpress/ClientApp/package.json]: Why were these dependencies added.

File: EventsExpress/ClientApp/src/components/category/category-edit.js [hunk 13]
```diff
@@ -7,6 +7,23 @@ import IconButton from "@material-ui/core/IconButton";
 
 class CategoryEdit extends Component {
 
+    componentDidMount = () => {
```
Comment (iuriikhrystiuk) [inline EventsExpress/ClientApp/src/components/category/category-edit.js]: Why is this file changed?

**Comment by VOLIMAX** [comment]
Forgot about code smells. Working on it now

File: EventsExpress/ClientApp/package.json [hunk 14]
```diff
@@ -70,6 +70,7 @@
     },
     "devDependencies": {
         "@types/leaflet": "1.5.19",
+        "history": "^5.0.0",
```
Comment (iuriikhrystiuk) [inline EventsExpress/ClientApp/package.json]: Is this dependency required?

**Review by iuriikhrystiuk — APPROVED** [review]
